### PR TITLE
Feature/spp 490/snapshot multi

### PR DESCRIPTION
**Pull request by sufyaniqbal**
# Description
Lambda file name change
https://collaborate2.ons.gov.uk/jira/browse/SPP-490

# How to test?
Pass a json message through the spp-es-takeon-egestion-spp490-db-export-input and check the lambda cloud watch log spp-es-takeon-egestion-spp490-export

Test env: spp490-snapshot
unit test

### CODE REVIEW

File: takeon-dbexport/main.go [hunk 1]
```diff
@@ -59,27 +59,48 @@ func handle(ctx context.Context, sqsEvent events.SQSEvent) error {
 		var messageJSON InputJSON
 		parseError := json.Unmarshal(messageDetails, &messageJSON)
 		if parseError != nil {
-			fmt.Printf("Error with JSON from db-export-input queue: %v", parseError)
+			sendToSqs("", "null", false)
```
Comment (dllmorgan) [inline takeon-dbexport/main.go:62]: Is this sending an null/empty message to the output queue?

**Review by dllmorgan — CHANGES_REQUESTED** [review]

File: takeon-dbexport/main_test.go [hunk 2]
```diff
@@ -26,6 +26,57 @@ func TestInputMessageReturnsExpected(t *testing.T) {
...
+
+
+func TestGetFileName_GivenMultipleSurveyPeriod_OutputCorrectFileName(t *testing.T){
+	var expected = "snapshot-066_201902-023_201902-023_201904-fwwekfnsdn"
+	var snapshotID = "fwwekfnsdn"
+	var singleSurveyPeriod = []SurveyPeriods {SurveyPeriods{Survey:"066", Period:"201902"}, SurveyPeriods{Survey:"023", Period:"201902"}, SurveyPeriods{Survey:"023", Period:"201904"} }
+	var actual, _ = getFileName(snapshotID, singleSurveyPeriod)
+
+	assert.EqualValues(t, expected, actual)
+}
```
Comment (dllmorgan) [inline takeon-dbexport/main_test.go:56]: These two tests need to check that no error has been returned.

**Review by dllmorgan — CHANGES_REQUESTED** [review]

**Review by larissanne — APPROVED** [review]
**Testing**

Correct JSON sent from input queue - Success:
Lambda logs show success

Invalid JSON sent from input queue - Fail:
Throws error in lambda logs

Extra digit in survey code from input queue - Fail:
Throws error in lambda logs

4 digit period from input queue i.e. for "201912" it was "1912" - Success:
Lambda logs show success

1 survey sent in JSON from input queue - Success:
Lambda logs show success

Period that doesn't exist - Fail:
Throws error in lambda logs

Empty JSON array sent from input queue for survey and period separately - Fail:
Throws error in lambda logs

Quotations need to be correct for message to be sent successfully and lambda to run successfuly

Timing was no more than 5 seconds for each scenario

**Review by dllmorgan — CHANGES_REQUESTED** [review]
re: The if statement inside getFileName (line 91) should be reversed. i.e. it is checking for an error condition and return the error message there and then. The rest of the function should be 'expected' behaviour. Having the usual return statement at the end of the function can be confusing for people to pick up.

**Review by dllmorgan — APPROVED** [review]

**Pull request merged**